app/view/button/SpatialQueryButton.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * This class is the button of cpsi mapview application that can be used to
+ * query an assigned layer by a draswn geometry. After the button if toggled,
+ * a draw interaction is activated on the map to define the query geometry.
+ *
+ * @class CpsiMapview.view.button.SpatialQueryButton
+ */
+Ext.define('CpsiMapview.view.button.SpatialQueryButton', {
+    extend: 'Ext.button.Button',
+    xtype: 'cmv_spatial_query_button',
+
+    requires: [
+        'CpsiMapview.model.button.SpatialQueryButton',
+        'CpsiMapview.controller.button.SpatialQueryButtonController'
+    ],
+
+    /**
+     * The viewModel for this class
+     */
+    viewModel: 'cmv_spatial_query_btn',
+
+    /**
+     * The controller for this class
+     */
+    controller: 'cmv_spatial_query_btn',
+
+    /**
+     * The name to be used e.g. in ComponentQueries
+     */
+    name: 'spatialQueryBtn',
+
+    /**
+     * The type of the geomety to graw / query with
+     * (Polygon, LineString, Point)
+     * Will be ignored if spatialOperator === 'bbox'
+     */
+    drawGeometryType: null,
+
+    /**
+     * Te type of spatial operato to use, see GeoExt.util.OGCFilter.
+     *                               topologicalOrSpatialFilterOperators
+     */
+    spatialOperator: null,
+
+    /**
+     * The name of the layer to query
+     */
+    queryLayerName: null,
+
+    /**
+     * The layer to query
+     */
+    queryLayer: null,
+
+    /**
+     * The drawn feature to query with
+     */
+    queryFeature: new ol.Collection(),
```

Comment:
Misleading name: Better `queryFeatures`. Since this is a collection I guess it can be more than one feature at a time?

Reply:
Currently it's not possible to draw / query multiple geometries since `OGCFilter` util of GeoExt3 doesn't support this. However, I agree that the name is misleading. Changed in 53ae2eb